From 4ab1075b9fc2353cd759ec62ae8bcfa0dfff14ca Mon Sep 17 00:00:00 2001
From: Aishwarya R <aisr@qti.qualcomm.com>
Date: Fri, 29 Apr 2022 12:34:13 +0530
Subject: [PATCH] ath12k: Disable 6 GHz Transmit power envelope for 320 MHZ

As transmit power envelope spec for 320 is not available yet
not filling power parameters for 320mhz channel width

Signed-off-by: Aishwarya R <aisr@qti.qualcomm.com>
---
 drivers/net/wireless/ath/ath12k/mac.c | 16 +++++++++++++++-
 1 file changed, 15 insertions(+), 1 deletion(-)

--- a/drivers/net/wireless/ath/ath12k/mac.c
+++ b/drivers/net/wireless/ath/ath12k/mac.c
@@ -12293,8 +12293,10 @@ ath12k_mac_op_assign_vif_chanctx(struct
 
 		arvif->chanctx = *ctx;
 
-		if (ahvif->vdev_type == WMI_VDEV_TYPE_STA)
+		if (ahvif->vdev_type == WMI_VDEV_TYPE_STA &&
+		    ctx->def.width != NL80211_CHAN_WIDTH_320) {
                         ath12k_mac_parse_tx_pwr_env(ar, vif, ctx);
+		}
         }
 
 	/* for some targets bss peer must be created before vdev_start */
